cassovary-core/src/test/scala/com/twitter/cassovary/graph/SemiSynchronizedDynamicGraphSpec.scala

```diff
@@ -0,0 +1,48 @@
+package com.twitter.cassovary.graph
+
+
+import org.scalatest.{Matchers, WordSpec}
+
+class SemiSynchronizedDynamicGraphSpec extends WordSpec with Matchers {
+  "A SemiSynchronizedDynamicDirectedGraphSpec" should {
+    "support adding nodes" in {
+      val graph = new SemiSynchronizedDynamicGraph()
+      for (i <- 0 until 3) {
+        graph.nodeCount shouldEqual i
+        graph.edgeCount shouldEqual 0
+        graph.getOrCreateNode(10 * i) // non-contiguous
+      }
+      graph.getOrCreateNode(10) // Accessing again should increase node count
+      graph.nodeCount shouldEqual 3
+      graph.existsNodeId(1000000) shouldEqual false
+    }
+
```

Comment:
Good idea.  I didn't add it before because I thought it would be hard to get the threads to actually interleave in an interesting way, but I added one now, especially since I wanted to test the new concurrent writing syncronization.  When I run the new test with print statements, the writers are definitely interleaving, and the reader occasionally interleaves (though on most runs the reader thread runs once before any writer, then doesn't run again until all the writers have finished).